In [2]:
from math import factorial as fl

In [3]:
def C(k,n):
    return int(fl(n) / (fl(k) * fl(n-k)))

In [4]:
def A(k,n):
    return int(fl(n) / (fl(n-k)))

In [16]:
def P(n):
    return int(fl(n))

Задача 1:

Группа студентов изучает 10 различных дисциплин. Сколькими способами можно составить расписание на понедельник, если в этот день должно быть 4 разных занятия?

Так как порядок предметов имеет значение, используем формулу для размещения:

In [17]:
A(4,10)

5040.0

Ответ: 5040 способами.

Задача 2:

Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

In [18]:
C(1,4)*C(3,48)+C(2,4)*C(2,48)+C(3,4)*C(1,48)+C(4,4)*C(0,48)

76145.0

Ответ: 76145 исходов.

Задача 3:
    
Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что два определённых человека будут сидеть рядом?

In [22]:
1/A(2,7) + 1/A(2,7)

0.047619047619047616

Задача 4:
    
Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

In [29]:
C(3,50) / C(3,60)

0.5727644652250146

In [30]:
(C(2,50) * C(1,10)) / C(3,60)

0.35797779076563413

Ответ: а) 0.57, б) 0.36

Задача 5:
    
Бросается игральная кость. Пусть событие A - появление чётного числа, событие B - появление числа больше трёх. Являются ли эти события независимыми?

Ответ: события независимые, так как появление четного числа не влияет на появление числа больше трех.

Задача 6:
    
Допустим, имеется некоторая очень редкая болезнь (поражает 0.1% населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99% больных этой болезнью и всего лишь в 1% случаев даёт ложный положительный ответ.
Вопрос: какова вероятность, что вы действительно больны ей?
Подсказка: вновь используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.

𝑃(𝐵) = 0.1 вероятность того, что тест положительный

𝑃(𝐴|𝐵) = 0.99 вероятность болезни, в случае положительного теста 

𝑃(𝐴)= 0.5 * 0,1 + 0.5 * 0.99

𝑃(𝐵|𝐴) = 𝑃(𝐵) * 𝑃(𝐴|𝐵) / 𝑃(𝐴)

𝑃(𝐵|𝐴)= (0.1 * 0.99) / (0.5 * 0,1 + 0.5 * 0.99)

In [43]:
0.1 * 0.99 / (0.5 * 0.1 + 0.5 * 0.99)

0.18165137614678897

Ответ: вероятность болезни 0.18